packages/app/src/cli/models/app/app.ts

```diff
@@ -20,13 +20,215 @@ export const LegacyAppSchema = zod
   .strict()
 
 // adding http or https presence and absence of new lines to url validation
-const validateUrl = (zodType: zod.ZodString) => {
+const validateUrl = (zodType: zod.ZodString, {httpsOnly = false, message = 'Invalid url'} = {}) => {
+  const regex = httpsOnly ? /^(https:\/\/)/ : /^(https?:\/\/)/
   return zodType
     .url()
-    .refine((value) => Boolean(value.match(/^(https?:\/\/)/)), {message: 'Invalid url'})
-    .refine((value) => !value.includes('\n'), {message: 'Invalid url'})
+    .refine((value) => Boolean(value.match(regex)), {message})
+    .refine((value) => !value.includes('\n'), {message})
 }
 
+const ensurePathStartsWithSlash = (arg: unknown) => (typeof arg === 'string' && !arg.startsWith('/') ? `/${arg}` : arg)
+const ensureHttpsOnlyUrl = validateUrl(zod.string(), {
+  httpsOnly: true,
+  message: 'Only https urls are allowed',
+})
+
+const SubscriptionEndpointUrlValidation = ensureHttpsOnlyUrl.optional()
+const PubSubProjectValidation = zod.string().optional()
+const PubSubTopicValidation = zod.string().optional()
+const ArnValidation = zod
+  .string()
+  .regex(
+    /^arn:aws:events:(?<aws_region>[a-z]{2}-[a-z]+-[0-9]+)::event-source\/aws\.partner\/shopify\.com(\.test)?\/(?<api_client_id>\d+)\/(?<event_source_name>.+)$/,
+  )
+  .optional()
+
+const WebhookSubscriptionSchema = zod.object({
+  topic: zod.string(),
+  sub_topic: zod.string().optional(),
+  format: zod.enum(['json', 'xml']).optional(),
+  include_fields: zod.array(zod.string()).optional(),
```

Comment:
I havent checked the API validation for these params yet but I have played around with it and it will let you pass anything you want in, but it will only return what you've requested if it is indeed valid. So I assumed we could just stick with that too